### PR TITLE
V2.10.0rc1

### PR DESCRIPTION
**Pull request by zklaus**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Review by valeriupredoi — APPROVED** [review]
looking spiffy, many thanks @zklaus :beer:

File: recipe/meta.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ build:
 requirements:
   host:
     - git
-    - python >=3.9
+    - python >=3.9,<3.12
```
Comment (bouweandela) [inline recipe/meta.yaml:23]: Bit sad about this, but I guess we can always build a new package once the issues upstream have been fixed.

**Review by bouweandela — APPROVED** [review]
:rocket:

File: recipe/meta.yaml (hunk 1 above)
Comment (zklaus) [reply recipe/meta.yaml:23]: Yeah. :shrug:

Comment (valeriupredoi) [reply recipe/meta.yaml:23]: that's gonna take two packages iterations, gents: once https://github.com/cedadev/MyProxyClient/pull/20 is merged we'll have to get a new version of MyProxyClient, then, with that, `esgf-pyclient` should work fine, but we'll have to unpin and rebuild - maybe for the main release? Although I got doubts Tool will work fine, I already know `aiohttp` is still cranky about 3.12

**Pull request merged**